src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -251,6 +254,37 @@ public RootNode doGetStudyTree(UUID studyId) {
         return root;
     }
 
+    @Transactional
+    public void copyStudyTree(AbstractNode nodeToDuplicate, UUID nodeParentId, UUID studyId) {
+        UUID rootId = null;
+        if (NodeType.ROOT.equals(nodeToDuplicate.getType())) {
+            StudyEntity tempStudy = new StudyEntity();
+            tempStudy.setId(studyId);
+            rootId = createRoot(tempStudy).getIdNode();
+        }
+        UUID referenceParentNodeId = rootId == null ? nodeParentId : rootId;
+
+        nodeToDuplicate.getChildren().stream().forEach(n -> {
+            UUID newModificationGroupId = UUID.randomUUID();
+            UUID nextParentId = null;
+
+            if (n instanceof NetworkModificationNode) {
+                NetworkModificationNode model = (NetworkModificationNode) n;
+                UUID modificationGroupToDuplicateId = model.getNetworkModification();
+                model.setNetworkModification(modificationGroupToDuplicateId != null ? newModificationGroupId : null);
+                model.setBuildStatus(modificationGroupToDuplicateId != null ? BuildStatus.BUILT_INVALID : BuildStatus.NOT_BUILT);
```

Comment:
Always set with BuildStatus.NOT_BUILT ?

Reply:
It'll be set to NOT_BUILD if there's no modification associated to the node, otherwise it's set to INVALID